File: app/src/main/java/com/cxsj/runhdu/TestActivity.java
package com.cxsj.runhdu;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.cxsj.runhdu.utils.ZipUtil;

public class TestActivity extends BaseActivity {

    private Button GzipButton;
    private Button zipButton;
    private EditText text;
    private TextView compressInfoText;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);
        setToolbar(R.id.toolbar_test, true);
        GzipButton = (Button) findViewById(R.id.GZip_compress_button);
        zipButton = (Button) findViewById(R.id.zip_compress_button);
        text = (EditText) findViewById(R.id.lab_text);
        compressInfoText = (TextView) findViewById(R.id.compress_info_text);
        compressInfoText.setText(String.format("压缩前：%d", text.length()));
        GzipButton.setOnClickListener(v -> {
            String str = text.getText().toString();
            String zipStr = ZipUtil.compress(str);
            text.setText(zipStr);
            compressInfoText.setText(String.format("压缩后%d", zipStr.length()));
            GzipButton.setClickable(false);
            zipButton.setClickable(false);
        });
        zipButton.setOnClickListener(v -> {
            String str = text.getText().toString();
            String zipStr = ZipUtil.compressForZip(str);
            text.setText(zipStr);
            compressInfoText.setText(String.format("压缩后%d", zipStr.length()));
            GzipButton.setClickable(false);
            zipButton.setClickable(false);
        });
        text.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {
                compressInfoText.setText(String.format("压缩前%d", s.toString().length()));
            }
        });
    }
}

File: app/src/main/java/com/cxsj/runhdu/TestActivity.kt
package com.cxsj.runhdu

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.util.Base64
import com.cxsj.runhdu.utils.MD5Util
import com.cxsj.runhdu.utils.ZipUtil
import kotlinx.android.synthetic.main.activity_test.*

class TestActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_test)
        setToolbar(R.id.toolbar_test, true)
        compress_info_text.text = "压缩前：${lab_text.length()}"

        GZip_button.setOnClickListener { _ ->
            val str = lab_text.text.toString()
            val zipStr = ZipUtil.compress(str)
            lab_text.setText(zipStr)
            compress_info_text.text = "压缩后：${zipStr.length}"
        }

        base64_button.setOnClickListener { _ ->
            val str = lab_text.text.toString()
            val base64Str = Base64.encode(str.toByteArray(), Base64.DEFAULT)
            lab_text.setText(String(base64Str))
            compress_info_text.text = "压缩后：${base64Str.size}"
        }

        GZip_decode_button.setOnClickListener { _ ->
            lab_text.setText(ZipUtil.decompress(lab_text.text.toString()))
        }

        base64_decode.setOnClickListener { _ ->
            lab_text.setText(String(Base64.decode(lab_text.text.toString(), Base64.DEFAULT)))
        }
        md5_button.setOnClickListener { _ ->
            lab_text.setText(MD5Util.encode(lab_text.text.toString()))
        }

        lab_text.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}

            override fun afterTextChanged(s: Editable) {
                compress_info_text.text = "压缩前：${s.toString().length}"
            }
        })
    }
}
